advanced/is-pattern-1/values.yaml

```diff
@@ -32,6 +32,9 @@ wso2:
     # Exported location on external NFS Server to be mounted at <IS_HOME>/repository/deployment/server/userstores
     sharedUserstores: ""
 
+  # The host for the ingress and HOST_NAME environment variable
+  host: "localhost"
```

Comment:
@rgauss shall we use a different hostname as the default? We do not recommend these Helm charts which are recommended for production grade deployments to be deployed in local Kubernetes environments such as, Minikube.

I would suggest a DNS name such as, `identity.wso2.com` as the default. This also eliminates the need set a hostname by default when deploying this chart with `helm install` command.

Further, this value needs to be included by default when adding customized keystores (see https://github.com/wso2/kubernetes-is/issues/224). This will be addressed as a future task.

Reply:
@chirangaalwis, IMHO production grade deployments shouldn't require end-users to modify their `/etc/hosts` file.

> I would suggest a DNS name such as, identity.wso2.com as the default. This also eliminates the need set a hostname by default when deploying this chart with helm install command.

I'm not sure what you mean.  That would only work for actually deploying to that host name.  Anyone else deploying the chart to their own domain will still need to set a host.

We could use something like `example.com`, but then it won't work out of the box for anyone.

Reply:
> I'm not sure what you mean. That would only work for actually deploying to that host name. Anyone else deploying the chart to their own domain will still need to set a host.

@rgauss sorry for not being clear. My concern is that the hostname `localhost` is not generally used in production grade deployments. It is most applicable for a local, evaluatory deployment.

IMO in a production environment, the best practice should be to set a custom DNS name with a mapping name server record for it.

We will evaluate this further and get back to you. Meanwhile, please feel free to drop in your thoughts, suggestions and concerns, if any.

Reply:
@chirangaalwis, indeed, in **any** non-local environment, production or otherwise, one **must** set a custom host, and the instructions should be very clear about that, but I'm not sure that means it can't have defaults that allow it to work for newcomers in common development environments.

Reply:
@chirangaalwis , @rgauss I think in terms of Ingress, we have a couple of concerns,

1.  Set an FQDN for the Ingress hostname(Ex: identity.example.com) as the default value which can be overridden via values.yaml
2. Propagate Ingress hostname configuration to the product configuration(deployment.toml).
3. Provide clear instructions on how to override the Ingress hostname and adding a DNS entry for the hostname(For evaluation, we can say put an entry in the /etc/hosts file).
4. Allow users to override the annotations of the Ingress. This is required if the user wants to use a different Ingress controller other than NGINX.

Reply:
@ragauss I assume you are setting **localhost**  as the default hostname for Ingress targeting users who are trying out this Helm chart locally. But if you look at a tool like Minikube, even, in that case, the user has to put an entry in the /etc/hosts file to point the Ingress controller running in the Minkube K8s cluster[1]. Please note that even in the use case that is described in the doc[1] uses an FQDN for the hostname. Therefore, it would be better if we can move forward with a default FQDN for the hostname. What do you think?

[1]https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/

Reply:
Hi @ThilinaManamgoda ,

> Set an FQDN for the Ingress hostname(Ex: identity.example.com) as the default value which can be overridden via values.yaml

Again, this ensures the chart won't work for anyone out of the box, and in general probably don't want to recommend overriding directly in values.yaml, rather a custom values override file or setting individual values.

> Propagate Ingress hostname configuration to the product configuration(deployment.toml).

Not sure what you mean here.  The `HOST_NAME` environment variable was already being set.  The purpose of the PR is not to change how and where host is configured in the deployment, just the value being provided.

> Provide clear instructions on how to override the Ingress hostname and adding a DNS entry for the hostname(For evaluation, we can say put an entry in the /etc/hosts file).

Are the README instructions not clear for overriding the ingress host?

Adding a true DNS entry is outside of the scope of the README, and editing a `/etc/hosts` file should not be recommended.  In many environments users aren't even permitted to do so.  Perhaps we can add an example on using nip.io.

> Allow users to override the annotations of the Ingress. This is required if the user wants to use a different Ingress controller other than NGINX.

That would definitely be outside of the scope of this PR and should be a separate issue.

Reply:
@ThilinaManamgoda 
> I assume you are setting localhost as the default hostname for Ingress targeting users who are trying out this Helm chart locally. But if you look at a tool like Minikube, even, in that case, the user has to put an entry in the /etc/hosts file to point the Ingress controller running in the Minkube K8s cluster[1]. Please note that even in the use case, that is described in the doc[1] uses an FQDN for the hostname. Therefore, it would be better if we can move forward with a default FQDN for the hostname. What do you think?

Docker for Desktop with support for Kubernetes is arguably more popular for local environments than Minikube these days, and ingress can be reached at localhost.

Again, I think nip.io examples would be a better option than suggesting editing `/etc/hosts`.

Reply:
@raguss The tool [nip.io](https://nip.io/) needs to be installed in the user's local environment. Which means this tool becomes a prerequisite for the Helm chart. I think that would affect the user experience, please correct me if I'm wrong.

Reply:
@ThilinaManamgoda, nip.io is not a tool to be installed, it's a free public service.  See [nip.io](https://nip.io/).